### ISSUE TITLE
Loss function is not squared in glove_cython?

### ISSUE THREAD

**Issue by ducovrossem** [maciejkula/glove-python#22]
Not sure if I am missing something here but thought I'd ask for clarification - the loss function is not squared.

```
loss = entry_weight * (prediction - c_log(count))
```

Also this implementation does not generate seperate vectors for when word is used in context?

**Comment by piskvorky** [comment]
FYI @maciejkula :
@dselivanov reports that this particular optimization (ignoring the context vectors) leads to a massive loss of accuracy:
http://rare-technologies.com/making-sense-of-word2vec/#comment-488

**Comment by maciejkula** [comment]
Interesting, I'll definitely have a look.

Incidentally, I think my more recent project (https://github.com/lyst/lightfm) should work really well on word embeddings (it uses a fancy learning-to-rank approach), I need to try it out.
